sys/net/network_layer/fib/fib.c

```diff
@@ -141,17 +140,35 @@ static int fib_find_entry(fib_table_t *table, uint8_t *dst, size_t dst_size,
             }
             else {
                 /* we try to find the most fitting prefix */
-                if ((ret_comp == 1)
-                    && (table->data.entries[i].global_flags & FIB_FLAG_NET_PREFIX)) {
-                    if ((prefix_size == 0) || (match_size > prefix_size)) {
+                if (ret_comp == 1) {
+                    if (table->data.entries[i].global_flags & FIB_FLAG_NET_PREFIX_MASK) {
+                        /* we shift the most upper flag byte back to get the number of prefix bits */
+                        size_t global_prefix_len = (table->data.entries[i].global_flags
+                                                    & FIB_FLAG_NET_PREFIX_MASK) >> FIB_FLAG_NET_PREFIX_SHIFT;
+
+                        if ((match_size >= global_prefix_len) &&
+                            ((prefix_size == 0) || (match_size > prefix_size))) {
```

Comment:
`prefix_size` is initialized as 0 and not set anywhere before to something else.